yb-voyager/src/tgtdb/eventQueue.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tgtdb
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+const (
+	QUEUE_DIR_NAME               = "queue"
+	QUEUE_SEGMENT_FILE_NAME      = "segment"
+	QUEUE_SEGMENT_FILE_EXTENSION = "ndjson"
+)
+
+type EventQueue struct {
+	QueueDirPath       string
+	SegmentNumToStream int64
+}
+
+func NewEventQueue(exportDir string) *EventQueue {
+	return &EventQueue{
+		QueueDirPath:       filepath.Join(exportDir, "data", QUEUE_DIR_NAME),
+		SegmentNumToStream: 0,
```

Comment:
So this implementation reads from the beginning of queue each time voyager runs??

Reply:
yeah its like that for now, will be modified after implementing resumability for importer